cypress/integration/core-opensearch-dashboards/opensearch-dashboards/left-navigation/left_navigation.spec.js

```diff
@@ -0,0 +1,314 @@
+      });
+    } else {
+      cy.visit('app/home');
+      validateMenuSection();
+    }
+  });
+
+  it('navigation should remember state of expand in browser', () => {
+    const validateMenuState = () => {
+      cy.get('.navToggleInLargeScreen').should('exist').click(); // expand the menu
+
+      cy.reload();
+      // navigation menu should remain expanded after reload
+      cy.get('.left-navigation-wrapper').within(() => {
+        isWorkspaceEnabled &&
+          cy.contains(`${workspaceName}_all`).should('exist');
+        cy.get('.navGroupEnabledHomeIcon').should('exist');
+        cy.get('input[type="search"]').should('exist');
+        cy.get('.bottom-container-expanded').should('exist');
+        cy.getElementByTestId('collapsibleNavShrinkButton')
+          .should('exist')
+          .click(); // collapse the menu
+      });
+
+      cy.reload();
+      // navigation menu should remain collapsed after reload
+      cy.get('.left-navigation-wrapper').find('.euiPanel').should('not.exist');
+      cy.get('.left-navigation-wrapper').within(() => {
+        isWorkspaceEnabled &&
+          cy.contains(`${workspaceName}_all`).should('not.exist');
+        cy.get('.navGroupEnabledHomeIcon').should('not.exist');
+        cy.get('input[type="search"]').should('not.exist');
+        cy.get('.bottom-container-expanded').should('not.exist');
+      });
+    };
+
+    if (isWorkspaceEnabled) {
+      createWorkspace('all').then(() => {
+        cy.visit(`w/${workspaceId}/app/all_overview`);
+        validateMenuState();
+      });
+    } else {
+      cy.visit('app/home');
+      validateMenuState();
+    }
+  });
+
+  it('validate navigation history functionality', () => {
+    const validateRecentHistory = () => {
+      // expand the navigation menu
+      cy.get('.navToggleInLargeScreen').should('exist').click();
+      cy.contains(/Visualizations/)
+        .should('exist')
+        .click();
+
+      let visualizationName;
+      cy.getElementByTestId('itemsInMemTable').within(() => {
+        cy.get('.euiLink')
+          .first()
+          .invoke('text')
+          .then((text) => {
+            visualizationName = text;
+            // Click to the first visualization item
+            cy.contains(visualizationName).click();
+          });
+      });
+
+      cy.wait(1000);
+
+      // open recent history dialog
+      cy.get('.headerRecentItemsButton').should('exist').click();
+      cy.get('div[role="dialog"]').within(() => {
+        // dialog displays correct visited content
+        cy.contains(/Recent assets/).should('exist');
+        cy.contains(visualizationName).should('exist');
+      });
+
+      // click recent history button again to close the dialog
+      cy.get('.headerRecentItemsButton').click();
+
+      // back to dashboard
+      cy.get('.left-navigation-wrapper').within(() => {
+        cy.contains(/Dashboards/)
+          .should('exist')
+          .click();
+      });
+
+      cy.wait(1000);
```

Comment:
Same here, wait may introduce flaky, please use assertion for consistent test result.

Reply:
Removed